base/abstractdict.jl

```diff
@@ -335,6 +335,75 @@ function _typeddict(d::AbstractDict, others::AbstractDict...)
     Dict{K,V}(d)
 end
 
+function reduce(::typeof(merge), dicts::AbstractVector{T}) where T <: AbstractDict{K,V} where {K,V}
+    isempty(dicts) && throw(ArgumentError("Cannot merge empty array"))
+    return _merge!(empty(first(dicts)),dicts) # use `empty`. `similar` makes routine 20x slower
+end
+
+function reduce(::typeof(merge), d::T, dicts::AbstractVector{T}) where T <: AbstractDict{K,V} where {K,V}
+    isempty(dicts) && return copy(d)
+    return _merge!(empty(first(dicts)),d,dicts) # There is no evidence that calling _merge! twice is less efficient.
+end
+
+# This catches typeof(T) == UnionAll.
+# The conversion of `dicts` to a vector of type Any makes this method much faster.
+function reduce(::typeof(merge), dicts::AbstractVector{T}) where T <: AbstractDict
+    isempty(dicts) && throw(ArgumentError("Cannot merge empty array"))
+    return reduce(merge, Any[x for x in dicts])
+end
+
+function reduce(::typeof(merge), d::AbstractDict, dicts::AbstractVector{T}) where T <: AbstractDict
+    isempty(dicts) && throw(ArgumentError("Cannot merge empty array"))
+    return reduce(merge, d, Any[x for x in dicts])
+end
+
+function reduce(m::typeof(merge), items::AbstractVector)
+    isempty(items) && throw(ArgumentError("Cannot merge empty vector"))
+    length(items) == 1 && return copy(first(items))
+    i1 = firstindex(items) + 1
+    i2 = lastindex(items)
+    return reduce(m,first(items), view(items,i1:i2))
+end
+
+function reduce(m::typeof(merge), d::AbstractDict, items::AbstractVector)
+    isempty(items) && return copy(d)
+    all(x -> isa(x,Dict), items) || return _reduce_dim(merge, d, items, Colon())
+    dictout = _typeddictA(d,items)
+    _merge!(dictout,items)
+end
+
+function _typeddictA(d::AbstractDict{K,V}, others) where {K,V}
+    k, v = K, V
+    for other in others
+        k = promote_type(k, keytype(other))
+        v = promote_type(v, valtype(other))
+    end
+    return Dict{k,v}(d)
+end
+
+function _merge!(dto::AbstractDict, d, dicts::AbstractArray)
+    _merge!(dto,d)
+    for dict in dicts
+        _merge!(dto,dict)
+    end
+    return dto
+end
+
+function _merge!(dto::AbstractDict, dicts::AbstractArray)
```

Comment:
This can be simplified to `_merge!(dto, dicts[1], dicts[2:end])`.

Reply:
Yes. But, shouldn't the last argument be a `view` ? I did benchmark the other use of `view` in this PR and the increased efficiency was not insignificant.

Reply:
Yeah, a view might be slightly better.